controller.go

```diff
@@ -181,6 +258,47 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	return network, nil
 }
 
+func (c *controller) newNetworkFromStore(n *network) {
+	c.Lock()
+	n.ctrlr = c
+	n.driver = c.drivers[n.networkType]
```

Comment:
> We need to handle this gracefully

Good point. Handled.

> We should not add the network into our local table if the driver doesn't exist

Agreed. Done

> We need to call driver.CreateNetwork when this happens.

Is there a value in calling the `driver.CreateNetwork` ? If the plugin/driver is multi-host aware, it will have its own mechanism to synchronize the network resource. Hence calling CreateNetwork isn't required. Also, if we have to provide that event, should we also give a hint to the driver that it is a remotely created network ?